### DISCUSSION TITLE
Connection is reset when receiving a payload that exceeds the max-packet-size limit?

### DISCUSSION THREAD

**Discussion by tsauerwein** [bytebeamio/rumqtt#856] [Q&A] [answered]
Hi! We noticed that if a payload is received that exceeds the configured "max-packet-size", the whole connection is [reset](https://github.com/bytebeamio/rumqtt/blob/c5bfdd74a535cadd518c12f7667e64f1ec68858c/rumqttc/src/v5/mqttbytes/v5/mod.rs#L340-L344).

Maybe it's due to our particular use-case, but with this behaviour our application would end up in a "re-connection loop" as long as the big payload is on the MQTT bus. Basically the steps would be:

1. MQTT connection is established
2. Subscribing to topics
3. Big payload is received -> disconnect
4. Go to 1

Did we understand the behaviour correctly? Is there an option not to disconnect in case of a `PayloadSizeLimitExceeded` error?

Thanks!

**Comment by swanandx** [comment]
Hey,
whether it tries to reconnect or not depend on how you are using rumqttc. Basically if you call `poll` it tries to connect.

>  Is there an option not to disconnect in case of a PayloadSizeLimitExceeded error?
>
as per protocol, it should disconnect! but if you don't want to disconnect, maybe just not configuring max-packet-size would solve your issue?

**Reply by tsauerwein** [reply]
Thanks for your reply!

> but if you don't want to disconnect, maybe just not configuring max-packet-size would solve your issue?

Is that possible? It seems there is a default of 10 * 1024 bytes.

**Reply by swanandx** [reply, marked as answer]
you can update the value using `set_max_packet_size`, but it looks like we will just use the [default value](https://github.com/bytebeamio/rumqtt/blob/main/rumqttc/src/v5/eventloop.rs#L285-L289) you mentioned!

Maybe we should let users specify `None`? for now, you can try to set it to some value large enough so your application won't crash?

thanks!

**Reply by tsauerwein** [reply]
Yes, that's what we are doing for now. Thanks for your feedback!
